### PR TITLE
[tests] Added smoke tests for conv solvers (ConvAsmImplicitGemmGTCDynamic*Xdlops and more). Added MIOPEN_DEBUG_TUNING_ITERATIONS_MAX. Some improvements and refactorings. 

### PR DESCRIPTION
**Pull request by atamazov**
### TODO
- [x] Add `GFX110X_ENABLED` where necessary.

### This PR adds smoke tests for conv solvers
- ConvAsmImplicitGemmGTCDynamic(Fwd|Bwd)Xdlops
- ConvBinWinograd3x3U
- ConvBinWinogradRxS
- ConvOclBwdWrW53
- ConvOclBwdWrW1x1
- fft
- ConvDirectNaiveConv*
- ConvAsm1x1U (with tuning)
- ConvAsm1x1UV2 (with tuning)
- ConvWinograd3x3MultipassWrW (3-3 only)
- ConvAsm3x3U (with tuning)

### By products:
- [tuning] Added `MIOPEN_DEBUG_TUNING_ITERATIONS_MAX`
  - This is to speed up the smoke testing of tunable solvers
- [tuning] Cleaner logging during tuning 
- [tests] Print datatype used for testing
- [NFC] Factor out some inline functions from generic_search.hpp
- [NFC] Fixed typo in the error message in ConvAsmImplicitGemmGTCDynamicWrwXdlopsNHWC.
- [NFC] Removed excessive IsGfx90aFp16altRequired() check from ConvAsm1x1UV2 because this solver does not support FP16

### CODE REVIEW

**Comment by atamazov** [comment]
TODO: Add `GFX110X_ENABLED` where necessary.

**Comment by atamazov** [comment]
[Attribution] @johnny-keker @junliume https://github.com/ROCmSoftwarePlatform/MIOpen/labels/testing https://github.com/ROCmSoftwarePlatform/MIOpen/labels/urgency_high because we have many merged PRs that refactor solvers but miss necessary tests. More solver tests are necessary to be added ASAP, but those depend on this PR.

Proposed reviewers: @averinevg @xinpin @shurale-nkn -- mostly to share knowledge.

**Comment by atamazov** [comment]
Ready for reviewing & CI testing.

File: src/solver/conv_asm_1x1u_stride2.cpp [hunk 1]
```diff
@@ -99,6 +99,14 @@ inline static bool IsLinear(const int v)
     return L <= v && v <= H;
 }
 
+static inline size_t divide_round_plus_inf(const size_t x, const unsigned y)
+{
+    assert(x >= 0 && y > 0);
```
Comment (junliume) [inline src/solver/conv_asm_1x1u_stride2.cpp]: @atamazov  like mentioned in the other PR, CI has some HIP tidy issue with this line:
```
[2023-01-11T01:44:14.744Z] /home/jenkins/workspace/MLLibs_MIOpen_PR-1898/src/solver/conv_asm_1x1u_stride2.cpp:104:14: warning: Unsigned expression 'x' can't be negative so it is unnecessary to test it. [unsignedPositive]
[2023-01-11T01:44:14.744Z]     assert(x >= 0 && y > 0);
[2023-01-11T01:44:14.744Z]              ^

Comment (atamazov) [reply src/solver/conv_asm_1x1u_stride2.cpp]: This assert is not necessary here. Leftover after moving from generic_search.hpp.

**Comment by atamazov** [comment]
@averinevg @xinpin @shurale-nkn @johnny-keker   Ready for reviewing & CI testing.

**Review by JehandadKhan — DISMISSED** [review]
@atamazov  Adding more logic and test cases to the CMakeLists.txt file means more and more work would be required in order to make tests package-able ? I would suggest we use the gtest machinery for all new added tests instead of piling on more technical debt ? 

What is your opinion?

**Comment by atamazov** [comment]
@JehandadKhan The most important part of the tests is convolution configs. The logic can be trivially moved (ported) to some other testing harness, e.g. if we have a counterpart of test_conv under gtest, then it should be easy to port smoke_solver_Conv* tests there. Please note that we need solver tests ASAP to prevent solvers from breaking during refactoring.

> ...instead of piling on more technical debt ?

Right now I can't not agree that this PR increases technical debt while adding more tests under gtest does not. But maybe I am missing something.

[Off-topic] Unfortunately I didn't take part in the decision making about gtest and I'm not yet sure that this decision is optimal. Maybe we have a discussion or ticket that lists the (relatively) new requirements for our tests? It would be even better if proposals, ideas etc were collected and analyzed there, so then I would just read it and understand what is needed and why... So far the only thing I know is that we want package-able tests, plus your comment at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1611#issuecomment-1194388099.

**Comment by averinevg** [comment]
> @averinevg @xinpin @shurale-nkn @johnny-keker Ready for reviewing & CI testing.

@atamazov Testing has been started

**Comment by averinevg** [comment]
CI passed

**Comment by atamazov** [comment]
@JehandadKhan I hope to receive your blessing after today's discussion.

**Comment by atamazov** [comment]
@JehandadKhan My hope is alive still

**Review by JehandadKhan — APPROVED** [review]
LGTM.

**Review by junliume — APPROVED** [review]

**Pull request merged**